### PR TITLE
Make failure in curl to fail Dockerfile.build.aarch64-linux image build

### PR DESCRIPTION
**Pull request by JanuszL**
- when curl command fails in the for loop inside
  Dockerfile.build.aarch64-linux the docker build silently continues
  instead of failing. This PR adds an `|| exit 1` after curl command
  to make sure that is fails verbosely

Signed-off-by: Janusz Lisiecki <jlisiecki@nvidia.com>

## Category:

**Bug fix** (*non-breaking change which fixes an issue*)

## Description:
- when curl command fails in the for loop inside
  Dockerfile.build.aarch64-linux the docker build silently continues
  instead of failing. This PR adds an `|| exit 1` after curl command
  to make sure that is fails verbosely

### Affected modules and functionalities:
- Dockerfile.build.aarch64-linux

### Key points relevant for the review:
- bash magic

### Tests
- [ ] Existing tests apply
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [x] N/A

### Documentation
- [ ] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [x] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [x] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

File: docker/Dockerfile.build.aarch64-linux [hunk 1]
```diff
@@ -84,7 +84,7 @@ RUN /bin/bash -c 'DALI_DEPS_VERSION_SHA=${DALI_DEPS_VERSION_SHA:-$(cat /tmp/DALI
 # hack - install cross headers in the default python paths, so host python3-config would point to them
 RUN export PYVERS="3.6.9 3.7.8 3.8.5 3.9.0 3.10.0" && \
     for PYVER in ${PYVERS}; do \
-        cd /tmp && curl -L https://www.python.org/ftp/python/${PYVER}/Python-${PYVER}.tgz | tar -xzf - && \
+        cd /tmp && curl -L https://www.python.org/ftp/python/${PYVER}/Python-${PYVER}.tgz | tar -xzf - || exit 1 && \
```
Comment (szalpal) [inline docker/Dockerfile.build.aarch64-linux:87]: Shouldn't there be braces around `||`? Like
```bash
(curl -L https://www.python.org/ftp/python/${PYVER}/Python-${PYVER}.tgz | tar -xzf - || exit 1)
```
But that's only an impression, I'm not bash aficionado ;)

**Review by szalpal — APPROVED** [review]

File: docker/Dockerfile.build.aarch64-linux (hunk 1 above)
Comment (JanuszL) [reply docker/Dockerfile.build.aarch64-linux]: Fixed

**Review by jantonguirao — APPROVED** [review]

File: docker/Dockerfile.build.aarch64-linux (hunk 1 above)
Comment (jantonguirao) [reply docker/Dockerfile.build.aarch64-linux]: It's actually the same effect:
```
> echo "A" && true || echo "C"
A

> echo "A" && false || echo "C"
A
C
```

Comment (JanuszL) [reply docker/Dockerfile.build.aarch64-linux]: It has but it @szalpal variant may improve the readability

**Pull request merged**